### PR TITLE
Add cos_e2e.py, test_utils and support for tensor inputs

### PR DESCRIPTION
**Pull request by phoenix-meadowlark**
- Add `test_utils` to dedupe jit comparison code and make testing multiple inputs easier
- Use `refjit.CompilerBackend` to make things a bit more obvious for the clueless.
- Add `TorchJitModuleInvoker` to allow `torch.Tensor`s to be passed to module invocations
- Add `cos_e2e.py` which is "failing" due to #135.

I did a cursory dig through the bindings for `capture_function` but am not familiar enough with C++ debug this on my own.

Adding `torch.cos` because I hit a number of walls trying to add `torch.diag`.

### CODE REVIEW

File: python/npcomp/compiler/pytorch/backend/refjit.py [hunk 1]
```diff
@@ -68,9 +84,9 @@ def compile(self, imported_module: Module):
         imported_module, refjit_backend.get_runtime_libs())
     return jit_module
 
-  def load(self, jit_module):
+  def load(self, jit_module) -> TorchJitModuleInvoker:
     """Loads a compiled artifact into the runtime.
 
-    Since this is a JIT instead of an AOT compiler,
+    Since this is a JIT instead of an AOT compiler, TODO: finish this sentence.
```
Comment (phoenix-meadowlark) [inline python/npcomp/compiler/pytorch/backend/refjit.py:90]: Please complete this sentence. I would but I can only think of really obvious things to say, like "...it compiles it just in time".

Comment (stellaraccident) [inline python/npcomp/compiler/pytorch/backend/refjit.py:90]: Fine to just remove the sentence (I can't remember what I was trying to convey).

File: frontends/pytorch/examples/mul_maximum_e2e.py [hunk 2]
```diff
@@ -8,29 +8,32 @@
...
 
-jit_result = jit_module.mul_maximum(lhs.numpy(), rhs.numpy(), threshold.numpy(),
-                                    bias.numpy())
-
-print(f"PyTorch Result = {result.numpy()}", file=sys.stderr)
-print(f"JIT Result = {jit_result}", file=sys.stderr)
-
-np.testing.assert_allclose(result.numpy(), jit_result)
+test_utils.compare_outputs(mul_maximum, jit_module.mul_maximum,
+                           lhs, rhs, threshold, bias,)
```
Comment (stellaraccident) [inline frontends/pytorch/examples/mul_maximum_e2e.py:37]: Remove trailing comma?

File: frontends/pytorch/examples/cos_e2e.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  result = torch.cos(input)
+  f.returns([result])
+
+backend = refjit.CompilerBackend()
+jit_module = backend.load(backend.compile(mb.module))
+
+logging.debug(f"Executing jit_module.cos")
+test_utils.compare_outputs(torch.cos, jit_module.cos, input)
+
+# This fails because the "Initial PyTorch IR" hardcodes the input value and
```
Comment (stellaraccident) [inline frontends/pytorch/examples/cos_e2e.py:32]: Can you file an issue?

**Review by stellaraccident — APPROVED** [review]

File: frontends/pytorch/examples/cos_e2e.py (hunk 3 above)
Comment (phoenix-meadowlark) [reply frontends/pytorch/examples/cos_e2e.py]: Done!

**Review by stellaraccident — APPROVED** [review]

**Pull request merged**